doc/reference/reference_lua/box_space/create_index.rst

```diff
@@ -113,321 +124,404 @@ On this page:
-
-    Functional indexes are indexes that call a user-defined function for forming
-    the index key, rather than depending entirely on the Tarantool default formation.
-    Functional indexes are useful for condensing or truncating or reversing or
-    any other way that users want to customize the index.
-
-    The function definition must expect a tuple (which has the contents of
-    fields at the time a data-change request happens) and must return a tuple
-    (which has the contents that will actually be put in the index).
-
-    The space must have a memtx engine. |br|
-    The function must be :ref:`persistent <box_schema-func_create_with-body>` and deterministic. |br|
-    The key parts must not depend on JSON paths. |br|
-    The ``create_index`` definition must include specification of all
-    key parts, and the function must return a table which has the
-    same number of key parts with the same types. |br|
-    The function must access key-part values by index, not by field name. |br|
-    Functional indexes must not be primary-key indexes. |br|
-    Functional indexes cannot be altered and the function cannot
-    be changed if it is used for an index, so the only way to change
-    them is to drop the index and create it again. |br|
-    Only sandboxed functions are suitable for functional indexes.
-
-    **Example:**
-
-    A function could make a key using only the first letter of a string field.
-
-    .. code-block:: lua
-
-        -- Step 1: Make the space.
-        -- The space needs a primary-key field, which is not the field that we
-        -- will use for the functional index.
         box.schema.space.create('x', {engine = 'memtx'})
         box.space.x:create_index('i',{parts={{field = 1, type = 'string'}}})
-        -- Step 2: Make the function.
-        -- The function expects a tuple. In this example it will work on tuple[2]
-        -- because the key source is field number 2 in what we will insert.
-        -- Use string.sub() from the string module to get the first character.
+
+#.  Make the function: the function expects a tuple. In this example it will
+    work on tuple[2] because the key source is field number 2 in what we will
+    insert. Use string.sub() from the string module to get the first character.
+
+    ..  code-block:: lua
+
         lua_code = [[function(tuple) return {string.sub(tuple[2],1,1)} end]]
-        -- Step 3: Make the function persistent.
-        -- Use the box.schema.func.create function for this.
+
+#.  Make the function persistent: use the box.schema.func.create function for this.
+
+    ..  code-block:: lua
+
         box.schema.func.create('F',
             {body = lua_code, is_deterministic = true, is_sandboxed = true})
-        -- Step 4: Make the functional index.
-        -- Specify the fields whose values will be passed to the function.
-        -- Specify the function.
+
+#.  Make the functional index: specify the fields whose values will be passed
```

Comment:
This is the first mention of this index, so we should use "a": "Make a functional index".